tensorflow/core/kernels/svd_op_gpu.cu.cc

```diff
@@ -0,0 +1,343 @@
+    OP_REQUIRES_OK(context, context->GetAttr("compute_uv", &compute_uv_));
+    OP_REQUIRES_OK(context, context->GetAttr("full_matrices", &full_matrices_));
+  }
+
+  void RunSVD(OpKernelContext* context, DoneCallback done, int64 m, int64 n,
+              int64 p, int64 batch_size, Scalar* input_ptr,
+              SScalar* outputS_ptr, Scalar* outputU_ptr, Scalar* outputVT_ptr,
+              int* dev_info_ptr, CudaSolver& solver) {
+    for (int64 i = 0; i < batch_size; ++i) {
+      int lda = m;
+      int ldu = m;
+      int ldvt = n;
+      Scalar* input = input_ptr + i * m * n;
+      SScalar* outputS = outputS_ptr + i * p;
+      Scalar* outputU = NULL;
+      Scalar* outputVT = NULL;
+      signed char jobu = 'N';
+      signed char jobvt = 'N';
+
+      if (compute_uv_) {
+        if (full_matrices_) {
+          outputU = outputU_ptr + i * m * m;
+          outputVT = outputVT_ptr + i * n * n;
+          jobu = 'A';
+          jobvt = 'A';
+        } else {
+          outputU = outputU_ptr + i * m * p;
+          outputVT = outputVT_ptr + i * n * p;
+          jobu = 'S';
+          jobvt = 'S';
+        }
+      }
+
+      OP_REQUIRES_OK_ASYNC(
+          context, solver.Gesvd(jobu, jobvt, m, n, input, lda, outputS, outputU,
+                                ldu, outputVT, ldvt, dev_info_ptr + i),
+          done);
+    }
+  }
+
+  void CheckResult(OpKernelContext* context, DoneCallback done,
+                   const std::vector<DeviceLapackInfo>& dev_info,
+                   CudaSolver& solver, Tensor& catch1, Tensor& catch2) {
+    auto info_checker = [context, dev_info, done, catch1, catch2](
+        const Status& status, const std::vector<HostLapackInfo>& /* unused */) {
+      Status full_status = status;
+      if (!full_status.ok()) {
+        full_status.Update(errors::InvalidArgument(kErrMsg));
+      }
+      OP_REQUIRES_OK_ASYNC(context, full_status, done);
+      done();
+    };
+
+    OP_REQUIRES_OK_ASYNC(context, solver.CopyLapackInfoToHostAsync(
+                                      dev_info, std::move(info_checker)),
+                         done);
+  }
+
+  // The SVD if m >= n
+  void PerformSVD_MgeqN(OpKernelContext* context, DoneCallback done, int64 m,
+                        int64 n, int64 p, const gtl::ArraySlice<int32>& perm,
+                        const Tensor& M, Tensor* S, Tensor* U, Tensor* V) {
+    TensorShape shapeRaw = M.shape();
+    shapeRaw.RemoveDim(shapeRaw.dims() - 1);
+    shapeRaw.RemoveDim(shapeRaw.dims() - 1);
+
+    // Transpose M, because cuSolver expects it to be column-major
+    TensorShape input_shape = shapeRaw;
+    input_shape.AddDim(n);
+    input_shape.AddDim(m);
+    Tensor input_copy;
+    OP_REQUIRES_OK_ASYNC(
+        context, context->allocate_temp(M.dtype(), input_shape, &input_copy),
+        done);
+    auto device = context->eigen_device<GPUDevice>();
+    DoTranspose(device, M, perm, &input_copy);
```

Comment:
You need to check the return value from DoTranpose. Wrap in OP_REQUIRES_OK macro.